expertise/service/expertise.py

```diff
@@ -853,35 +853,48 @@ async def worker_process(self, job, token):
 
         try:
             self.logger.info(f"In polling worker...")
-            job_completed, job_error = False, False
-
             for attempt in range(self.max_attempts):
-                self.logger.info(f"In attempt {attempt + 1} of {self.max_attempts}...")
+                self.logger.info(f"{redis_id} - attempt {attempt + 1} of {self.max_attempts}...")
                 status = self.cloud.get_job_status_by_job_id(user_id, cloud_id)
                 self.logger.info(f"Invoked get_job_status_by_job_id for {redis_id} - status: {status}")
 
-                # Set status in Redis
-                config = self.redis.load_job(redis_id, user_id)
-                self.update_status(config, status['status'], status['description'])
-
-                if status['status'] == JobStatus.COMPLETED:
-                    self.logger.info(f"Job {redis_id} completed.")
-                    job_completed = True
-                    break
-                elif status['status'] == JobStatus.ERROR:
-                    self.logger.info(f"Job {redis_id} encountered an error.")
-                    job_completed = False
-                    job_error = True
-                    break
-
-                self.logger.info(f"Job {redis_id} status: {status['status']}. Retrying in {self.poll_interval} seconds...")
+                # Check status validity
+                self.logger.info(f"INFO: before status check")
+                if status and isinstance(status, dict) and 'status' in status and 'description' in status:
+                    self.logger.info(f"INFO: after status check")
+                    config = self.redis.load_job(redis_id, user_id)
```

Comment:
we don't need to do this in this PR, but I suggest we get this information from the job of the queue